erpnext/accounts/doctype/purchase_invoice/purchase_invoice.js

```diff
@@ -520,4 +520,12 @@ frappe.ui.form.on("Purchase Invoice", {
 		}
 		frm.toggle_reqd("supplier_warehouse", frm.doc.is_subcontracted==="Yes");
 	}
-})
\ No newline at end of file
+})
+frappe.ui.form.on('Purchase Invoice', {
+	cost_center: function(frm, cdt, cdn) {
+		erpnext.utils.copy_parent_value_in_all_row(frm.doc, cdt, cdn, "items", "cost_center", "cost_center");
```

Comment:
Why copying parent cost center to item rows? Expenses can be booked against multiple cost center and it should be based on item's default cost center.

Reply:
The purpose of copying parent cost center to item is to book purchase against cost center selected in parent document to maintain cost center wise payable.

Reply:
Once again, my comment here of being able to see the total payable even when viewing CC wise accounts. The idea is that we would CC for product lines similar to classes in QBO so same branch may have more than one CCs. This is the scenario why visibility of both CC-level and total outstanding data would be useful.

Reply:
Let's don't copy parent CC to child items. Keep them separate. We have a plan to introduce "Department" and "Branch" inside item table, so that we can get P&L analysis based on branch / department.

Reply:
> Let's don't copy parent CC to child items. Keep them separate. We have a plan to introduce "Department" and "Branch" inside item table, so that we can get P&L analysis based on branch / department.

Okay, I will remove function to copy parent CC to child.